cmd/clusterctl/client/config/reader_viper.go

```diff
@@ -73,13 +78,27 @@ func (v *viperReader) Init(path string) error {
 	viper.AllowEmptyEnv(true)
 	viper.AutomaticEnv()
 
-	// Reads the clusterctl config file
 	if path != "" {
-		if _, err := os.Stat(path); err != nil {
-			return err
+		switch {
+		case strings.Index(path, "http://") == 0 || strings.Index(path, "https://") == 0:
```

Comment:
What about relying on url.Scheme instead of on string parsing (see e.g. https://github.com/kubernetes-sigs/cluster-api/blob/8f0485227dff19064f6625cd912b53878a464130/cmd/clusterctl/client/repository/client.go#L154-L180)

Reply:
done